src/index.js

```diff
@@ -2,10 +2,34 @@ import React from 'react';
 import ReactDOM from 'react-dom';
 import 'normalize.css/normalize.css';
 
-import App from './components/App';
-import './firebase/firebase';
+import { history } from 'router/AppRouter';
+import { LOGIN_PAGE_ROUTE, ACTIVITY_PLAN_ROUTE } from 'constants/routes';
+import { login, logout } from 'store/auth/actions';
 
-ReactDOM.render(<App />, document.getElementById('app-root'));
+import App, { store } from './components/App';
+import { firebase } from './firebase/firebase';
+
+let hasRendered = false;
+const renderApp = () => {
+  if (!hasRendered) {
+    ReactDOM.render(<App />, document.getElementById('app-root'));
+    hasRendered = true;
+  }
+};
+
+firebase.auth().onAuthStateChanged(user => {
+  if (user) {
+    store.dispatch(login(user.uid));
+    renderApp();
+    if (history.location.pathname === LOGIN_PAGE_ROUTE) {
+      history.push(ACTIVITY_PLAN_ROUTE);
+    }
+  } else {
+    store.dispatch(logout());
```

Comment:
why there is an logout? I mean, if there is no user, there is no one to be logout 🤔

Reply:
There is no login in firebase (as you said no one is loged in) but this logout() action removes login from store and app is rendered properly.